interfaces/machine_parameters/ring.py

```diff
@@ -0,0 +1,588 @@
+            self.Particle.mass,
+            self.circumference_design, sample_func,
+            interpolation, start, stop,
+            store_turns=store_turns)
+
+        # Getting the cycle time from the interpolation
+        self.cycle_time = momentum_processed[1]
+
+        # The machine turn numbers corresponding to the cycle_time
+        # are kept if the store_turns is enabled
+        if store_turns:
+            self.parameters_at_turn = self._parameters_at_turn
+            self.use_turns = momentum_processed[0].astype(int)
+        else:
+            self.parameters_at_turn = self._no_parameters_at_turn
+            self.use_turns = momentum_processed[0]
+
+        # Updating the number of turns in case it was changed after ramp
+        # interpolation
+        self.n_turns = momentum_processed.n_turns
+
+        # Storing the momentum program and computing all associated values
+        self.momentum = momentum_processed[2:]
+        self.beta = rt.mom_to_beta(
+            momentum_processed[2:], self.Particle.mass)
+        self.gamma = rt.mom_to_gamma(
+            momentum_processed[2:], self.Particle.mass)
+        self.energy = rt.momentum_to_energy(
+            momentum_processed[2:], self.Particle.mass)
+        self.kin_energy = rt.momentum_to_kin_energy(
+            momentum_processed[2:], self.Particle.mass)
+
+        # Extracting and combining the orbit length programs
+        self.section_length = ring_programs.orbit_length_program.combine_single_sections(
+            *[section.length for section in self.Section_list],
+            interpolation='linear')
+
+        # Reshaping to match the dimensions of the synchronous data program
+        self.section_length = self.section_length.reshape(
+            self.n_sections, self.cycle_time, self.use_turns)
+
+        # Getting the circumference and radius (including potential orbit
+        # bumps)
+        self.circumference = np.sum(self.section_length, axis=0)
+        self.radius = self.circumference / (2 * np.pi)
+
+        # Computing the revolution period on the design orbit
+        # as well as revolution frequency and angular frequency
+        self.t_rev_design = np.dot(self.section_length_design,
+                                   1 / (self.beta * c))
+        self.f_rev_design = 1 / self.t_rev_design
+        self.omega_rev_design = 2 * np.pi * self.f_rev_design
+
+        # Computing the time of flight in each section
+        # and the revolution period on the beam orbit including
+        # possible orbit bumps
+        self.t_rev = np.sum(np.array(
+            self.section_length / (self.beta * c)), axis=0)
+        self.f_rev = 1 / self.t_rev
+        self.omega_rev = 2 * np.pi * self.f_rev
+
+        # Recalculating the delta_E
+        self.delta_E = np.diff(self.energy, axis=1)
+        if self.n_turns > len(self.use_turns):
+            self.delta_E = np.zeros(self.energy.shape)
+            self._recalc_delta_E()
```

Comment:
Looks a little awkward to me, maybe better as:
```suggestion
        if self.n_turns > len(self.use_turns):
            self.delta_E = np.zeros(self.energy.shape)
            self._recalc_delta_E()
        else:
            self.delta_E = np.diff(self.energy, axis=1)
```

Reply:
Corrected